### PR TITLE
Configure Brave Ads next payment date from variations seed

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/14067

## Submitter Checklist:

- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)
- [ ] Requested a security/privacy review as needed

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/features/ad_rewards/ad_rewards_features.cc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+const char kFeatureName[] = "AdRewards";
+const char kFieldTrialParameterNextPaymentDay[] = "next_payment_day";
+
+const int kDefaultNextPaymentDay = 5;
+
+}  // namespace
+
+const base::Feature kAdRewards{kFeatureName, base::FEATURE_ENABLED_BY_DEFAULT};
+
+bool IskAdRewardsEnabled() {
```
Comment (moritzhaller) [inline vendor/bat-native-ads/src/bat/ads/internal/features/ad_rewards/ad_rewards_features.cc:24]: typo, should be `IsAdRewardsEnabled` ?

File: vendor/bat-native-ads/src/bat/ads/internal/features/ad_rewards/ad_rewards_features.cc [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  return base::FeatureList::IsEnabled(kkAdRewards);
+}
+
+int GetkAdRewardsNextPaymentDay() {
```
Comment (moritzhaller) [inline vendor/bat-native-ads/src/bat/ads/internal/features/ad_rewards/ad_rewards_features.cc:28]: typo, should be `GetAdRewardsNextPaymentDay`?

**Review by moritzhaller — APPROVED** [review]
LGTM++

File: vendor/bat-native-ads/src/bat/ads/internal/features/ad_rewards/ad_rewards_features.cc (hunk 1 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/features/ad_rewards/ad_rewards_features.cc]: Fixed

File: vendor/bat-native-ads/src/bat/ads/internal/features/ad_rewards/ad_rewards_features.cc (hunk 2 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/features/ad_rewards/ad_rewards_features.cc]: Fixed

**Comment by tmancey** [comment]
CI failed due to BraveShieldsAPIBrowserTest.AllowScriptsOnceIframe (../../brave/browser/extensions/api/brave_shields_api_browsertest.cc:185) which is a known issue

**Pull request merged**